api/src/services/replication-limit-log.js

```diff
@@ -0,0 +1,93 @@
+const moment = require('moment');
+const db = require('../db');
+const logger = require('../logger');
+
+const LOG_TYPES = {
+  replicationCount: 'replication-count-'
+};
+const LOG_COUNT_DIFF = 100;
+const LOG_MONTH_DIFF = 1;
+
+const persistLog = (info) => {
+  if (!info || !info.user || !info.user.user_name) {
+    const error = new Error('Error when persisting log: Log Information missing.');
+    return Promise.reject(error);
+  }
+
+  const logDocId = LOG_TYPES.replicationCount + info.user.user_name;
+
+  return db.medicLogs
+    .get(logDocId)
+    .catch(error => {
+      if (error.status === 404) {
+        return { _id: logDocId };
+      }
+      throw error;
+    })
+    .then(doc => {
+      if (!isLogDifferent(doc, info)) {
+        return;
+      }
+
+      const logDoc = Object.assign(doc, info);
+      return db.medicLogs.put(logDoc);
+    });
+};
+
+const isLogDifferent = (oldLog, newLog) => {
+  if (!oldLog.log_date && !oldLog.count) {
+    return true;
+  }
+
```

Comment:
Check the countDiff here first and return early if true to save us doing relatively expensive date logic unnecessarily.

Reply:
Fixed